### PR TITLE
Added Default trait for Cow.

### PR DESCRIPTION
**Pull request by XAMPPRocky**
Adds a default implementation for Cow. Which is the Owned's default.

### CODE REVIEW

**Comment by alexcrichton** [comment]
Thanks for the PR! Could you separate out the formatting changes from the `Default` addition? Doing so as a separate commit should be fine to do so as well.

cc @rust-lang/libs, any objections?

File: src/libcollections/borrow.rs [hunk 1]
```diff
@@ -248,8 +261,20 @@ impl<'a, B: ?Sized> fmt::Display for Cow<'a, B>
     }
 }
 
+#[stable(feature = "default", since = "1.11.0")]
+impl<'a, B: ?Sized> Default for Cow<'a, B>
+    where B: Default + ToOwned,
```
Comment (ollie27) [inline src/libcollections/borrow.rs]: Does `B` really need to implement `Default` as well?

**Comment by Aaronepower** [comment]
@alexcrichton Sorry, left rustfmt on. Fixed it.

File: src/libcollections/borrow.rs (hunk 1 above)
Comment (Aaronepower) [inline src/libcollections/borrow.rs]: Yeah, I guess it doesn't only it's Owned version does.

**Comment by brson** [comment]
Seems consistent with our trend of adding more defaults. Please [squash](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html) these commits though.

**Comment by Aaronepower** [comment]
@brson Done.

**Comment by brson** [comment]
Thanks @Aaronepower. Just waiting for the libs team to confer about it.

**Comment by alexcrichton** [comment]
Discussed during libs triage yesterday the conclusion was to merge, thanks for the PR @Aaronepower!

@bors: r+ 703d7b519e69daf03e34edcd1af2957e393414b5

**Pull request merged**